envs/environment.yaml

```diff
@@ -8,11 +8,11 @@ channels:
 - bioconda
 - gurobi
 dependencies:
-- python>=3.8
+- python>=3.8, <3.12  # 3.12 has weird issues with snakemake https://github.com/snakemake/snakemake/issues/2480
 - pip
 - mamba   # esp for windows build
 
-- pypsa>=0.25
+- pypsa>=0.25, <=0.30.3
```

Comment:
Would be great to add a comment on the reason on this restriction also here.

Reply:
I create a new branch for testing >=v0.31.
I did this in the past and there was a clustering issue that was introduced and fixed some versions after. I cannot pinpoint the exact error though.
In the new iteration, we will surely need to skip few pypsa versions and while we are there, we may also update snakemake (maybe)

Reply:
This is the issue:
https://github.com/pypsa-meets-earth/pypsa-earth/actions/runs/15229922110/job/42836546628?pr=1460

but not sure where it is solved; there is the need to do some testing for future releases

Reply:
Ahh, interesting! Many thanks for investigating this.

I have been thinking about the known issues with PyPSA. There has been a version which was not recommended for usage due to some bug. I think it has been [this issue](https://github.com/PyPSA/PyPSA/issues/1144) with statistics introduced in v0.32.0, fixed in [v0.34.0](https://github.com/PyPSA/PyPSA/pull/1172) by [this PR](https://github.com/PyPSA/PyPSA/pull/1172). Sorry for not being clear.